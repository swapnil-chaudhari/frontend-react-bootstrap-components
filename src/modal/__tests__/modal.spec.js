import Modal from '../modal';
import React from 'react';
import renderShallow from 'render-shallow';
import { findWithClass } from 'react-shallow-testutils';

describe('<Modal>', () => {

    let props;
    let component;
    let onRequestCloseSpy;

    beforeEach(() => {
        onRequestCloseSpy = jest.fn();
        props = {
            className: 'MyClass',
            isOpen: true,
            onRequestClose: onRequestCloseSpy
        };

    });

    describe('when rendered with all props', () => {
        beforeEach(() => {
            component = renderShallow(<Modal { ...props } >Some stuff</Modal>).output;
        });

        test('renders <ReactModal> with the right props', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when passed a value for `shouldCloseOnOverlayClick`', () => {
        beforeEach(() => {
            props.shouldCloseOnOverlayClick = false;
            component = renderShallow(<Modal { ...props } >Some stuff</Modal>).output;
        });

        test('renders <ReactModal> with the right props', () => {
            expect(component.props.shouldCloseOnOverlayClick).toEqual(false);
        });
    });

    describe('when zindex is passed', () => {
        beforeEach(() => {
            props.zIndex = 5;
            component = renderShallow(<Modal { ...props } >Some stuff</Modal>).output;
        });

        test('renders <ReactModal> with the right props', () => {
            expect(component.props.style.overlay.zIndex).toEqual(5);
        });
    });

    describe('when passed a value for `headerLabel`', () => {
        beforeEach(() => {
            props.headerLabel = 'Billy Yo';
            component = renderShallow(<Modal { ...props } >Some stuff</Modal>).output;
        });

        test('renders <ReactModal> with the right props', () => {
            findWithClass(component, 'ReactModal-close').props.onClick();
            expect(component).toMatchSnapshot();
            expect(props.onRequestClose).toHaveBeenCalledWith();
        });
    });

    describe('when passed a value for `footer`', () => {
        beforeEach(() => {
            props.footer = <div className="yummy">Hi</div>;
            component = renderShallow(<Modal { ...props } >Some stuff</Modal>).output;
        });

        test('renders with a footer', () => {
            const footer = findWithClass(component, 'ReactModal-footer');
            expect(footer).toMatchSnapshot();
        });
    });
});
