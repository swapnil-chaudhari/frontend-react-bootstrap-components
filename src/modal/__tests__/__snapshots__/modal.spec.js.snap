// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal> when passed a value for \`footer\` renders with a footer 1`] = `
<div
  className="ReactModal-footer"
>
  <div
    className="yummy"
  >
    Hi
  </div>
</div>
`;

exports[`<Modal> when passed a value for \`headerLabel\` renders <ReactModal> with the right props 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal-body-open"
  className="MyClass ReactModal"
  closeTimeoutMS={0}
  contentLabel="Modal"
  isOpen={true}
  onRequestClose={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
    }
  }
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "overlay": Object {
        "backgroundColor": "rgba(75, 75, 75, 0.74902)",
        "zIndex": 3,
      },
    }
  }
>
  <div
    className="ReactModal-content"
  >
    <header
      className="ReactModal-header"
    >
      <div
        className="ReactModal-close"
        onClick={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
          }
        }
      >
        <Close />
      </div>
      <h4
        className="ReactModal-header-text"
      >
        Billy Yo
      </h4>
    </header>
    <div
      className="ReactModal-body"
    >
      Some stuff
    </div>
  </div>
</Modal>
`;

exports[`<Modal> when rendered with all props renders <ReactModal> with the right props 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal-body-open"
  className="MyClass ReactModal"
  closeTimeoutMS={0}
  contentLabel="Modal"
  isOpen={true}
  onRequestClose={[MockFunction]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "overlay": Object {
        "backgroundColor": "rgba(75, 75, 75, 0.74902)",
        "zIndex": 3,
      },
    }
  }
>
  <div
    className="ReactModal-content"
  >
    <div
      className="ReactModal-body"
    >
      Some stuff
    </div>
  </div>
</Modal>
`;
